File: app/src/main/java/com/sadashi/apps/ui/material/views/adapter/SampleListAdapter.java
package com.sadashi.apps.ui.material.views.adapter;

import android.content.Context;
import android.support.annotation.NonNull;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import com.sadashi.apps.ui.material.R;

import butterknife.BindView;
import butterknife.ButterKnife;

public class SampleListAdapter extends RecyclerView.Adapter<SampleListAdapter.ViewHolder> {

    class ViewHolder extends RecyclerView.ViewHolder {
        @BindView(R.id.text)
        TextView text;

        ViewHolder(View itemView) {
            super(itemView);
            ButterKnife.bind(this, itemView);
        }
    }

    private LayoutInflater inflater;

    public SampleListAdapter(@NonNull Context context) {
        this.inflater = (LayoutInflater) context.getSystemService(Context.LAYOUT_INFLATER_SERVICE);
    }

    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View v = inflater.inflate(R.layout.sample_list_item, parent, false);
        return new ViewHolder(v);
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolder holder, int position) {
        holder.text.setText("hoge");
    }

    @Override
    public int getItemCount() {
        return 50;
    }

}

File: app/src/main/java/com/sadashi/apps/ui/material/views/adapter/SampleListAdapter.kt
package com.sadashi.apps.ui.material.views.adapter

import android.support.v7.widget.RecyclerView
import android.view.View
import android.view.ViewGroup
import com.sadashi.apps.extensions.inflate
import com.sadashi.apps.ui.material.R
import kotlinx.android.synthetic.main.sample_list_item.view.*

class SampleListAdapter : RecyclerView.Adapter<SampleListAdapter.ViewHolder>() {

    class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        fun bind() {
            itemView.text.text = "dummy"
        }
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder =
        ViewHolder(parent.inflate(R.layout.sample_list_item))

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        holder.bind()
    }

    override fun getItemCount(): Int = 50
}
